mpa/modules/ov/graph/parsers/cls/cls_base_parser.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from typing import Dict, List, Optional
+
+from mpa.utils.logger import get_logger
+
+from ..builder import PARSERS
+from ..parser import parameter_parser
+
+
+logger = get_logger()
+
+
+@PARSERS.register()
+def cls_base_parser(
+    graph, component: str = "backbone"
+) -> Optional[Dict[str, List[str]]]:
+    assert component in ["backbone", "neck", "head"]
+
+    possible_neck_input_types = ["ReduceMean", "MaxPool", "AvgPool"]
+
+    result_nodes = graph.get_nodes_by_types(["Result"])
+    if len(result_nodes) != 1:
+        logger.debug("More than one reulst nodes are found.")
+        return None
+    result_node = result_nodes[0]
+
+    neck_input = None
+    for _, node_to in graph.bfs(result_node, True, 20):
+        if node_to.type in possible_neck_input_types:
+            logger.debug(f"Found neck_input: {node_to.name}")
+            neck_input = node_to
+            break
+
+    if neck_input is None:
+        logger.debug("Can not determine the output of backbone.")
+        return None
+
+    part_of_neck_types = [
+        "Reshape",
+        "Squeeze",
+        "Unsqueeze",
+        "Concat",
+        "Convert",
+        "ShapeOf",
+        "StridedSlice",
+        "Transpose",
+    ]
```

Comment:
Isn't it good to move this to global variable at the top of the code?

Reply:
how about renaming this to `possible_neck_types`

Reply:
Fixed in https://github.com/openvinotoolkit/model_preparation_algorithm/pull/95/commits/e173066e7d7a43670452fb1b3dc9e99690c4503a